frontend/components/form/FormField.vue

```diff
@@ -0,0 +1,108 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+
+interface Props {
+    fieldKey: string;
+    fieldConfig: {
+        type: string;
+        title: string;
+        description: string;
+        enum?: string[];
+        items?: {
+            type: string;
+            title: string;
+            uniqueItems?: boolean;
+        }
+    };
+    modelValue: string | string[];
+    error?: string;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    error: ''
+});
+
+const emit = defineEmits<{
+    (e: 'update:modelValue', value: string | string[]): void;
+}>();
+
+const isSelectField = props.fieldConfig.enum !== undefined;
+const isArrayField = props.fieldConfig.type === 'array';
+const arrayValues = ref<string[]>(Array.isArray(props.modelValue) ? props.modelValue : [props.modelValue || '']);
+
+const addArrayItem = () => {
+    arrayValues.value.push('');
+    emit('update:modelValue', arrayValues.value);
+};
+
+const deleteArrayItem = (index: number) => {
+    arrayValues.value.splice(index, 1);
+    emit('update:modelValue', arrayValues.value);
+};
+
+const updateArrayItem = (index: number, value: string) => {
+    arrayValues.value[index] = value;
+    emit('update:modelValue', arrayValues.value);
+};
+
+
+// Base classes for inputs
+const inputBaseClasses = `
```

Comment:
We should be careful with that because tailwind may not compile the classes if they are not in the template. 
Instead you could create a CSS class which use the [@apply](https://tailwindcss.com/docs/functions-and-directives) directive to group classes into a single one